packages/common/src/EntityVertex.ts

```diff
@@ -0,0 +1,92 @@
+import { Entity } from "./Entity";
+import { TextBox } from "./TextBox";
+import { d3SelectionType } from "./Widget";
+
+export class EntityVertex extends Entity {
+    protected _textbox_widget: TextBox;
+    protected _element_textbox: d3SelectionType;
+    constructor() {
+        super();
+        this._textbox_widget = new TextBox();
+    }
+    enter(domNode, element) {
+        this._element_textbox = element.append("g").attr("class", "entity_textbox");
+        super.enter(domNode, element);
+        this._textbox_widget.target(this._element_textbox.node());
+    }
+    update() {
+        super.update.apply(this, arguments);
+        this._textbox_widget
+            .text(this._title_widget.text());
+        this._icon_widget
+            .shape_colorFill(this.iconColorFill())
+            .shape_colorStroke(this.iconColorStroke());
+        this._background_widget
+            .colorFill("none")
+            .colorStroke("none");
+
+        this._icon_widget.render();
+        this._textbox_widget.render();
+        const icon_bbox = this._icon_widget.getBBox(true);
+        const text_bbox = this._textbox_widget.getBBox(true);
+        let _icon_x = 0;
+        let _icon_y = -(text_bbox.height / 2) - (icon_bbox.height / 3);
+        switch (this.iconAnchor()) {
+            case "start":
+                _icon_x = -(text_bbox.width / 2) + (icon_bbox.width / 3);
+                break;
+            case "end":
+                _icon_x = (text_bbox.width / 2) - (icon_bbox.width / 3);
+                break;
+            case "left":
+                _icon_x = -(text_bbox.width / 2) - (icon_bbox.width / 2);
+                _icon_y = 0;
+                break;
+        }
+        this._icon_widget.move({
+            x: _icon_x,
+            y: _icon_y
+        });
+        let anno_w_sum = 0;
+        const anno_gap = this.annotationSpacing();
+        for (const i in this._annotation_widgets) {
+            const anno_bbox = this._annotation_widgets[i]._bbox;
+            this._annotation_widgets[i].move({
+                x: (text_bbox.width / 2) - (anno_bbox.width / 2) - anno_w_sum + anno_gap,
+                y: (text_bbox.height / 2) + (anno_bbox.height / 2) - anno_gap
+            });
+            anno_w_sum += anno_bbox.width + anno_gap;
+        }
```

Comment:
You should have a common function for this which positions the row of annotations at a given place.